### DISCUSSION TITLE
Refactor support for moving object definitions between modules

### DISCUSSION THREAD

**Discussion by foster999** [microsoft/pylance-release#2659] [Enhancement]
I would like the refactor menu to include an option to move the selected object(s) definitions to another Python file.

This would provide the most benefit in package development where, similar to microsoft/pylance-release#2283, this action should update usage and imports of this object throughout the current project. This functionality is exhibited by [PyCharm's move refactoring](https://www.jetbrains.com/help/pycharm/move-refactorings.html).

**Comment by debonte** [comment]
Thanks for the feature request. We are looking into this.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by benlindsay** [comment]
This would be an amazing feature to add. I'm guessing there would be more requests for this if more people were aware that it was a possibility. The refactoring support for moving files around within a python module is fantastic, and it would be great if that would extend to moving functions or classes around between files.

**Comment by judej** [comment]
This is a dupe of https://github.com/microsoft/pylance-release/discussions/2620
